frontend/src/components/MatchedRoom.js

```diff
@@ -1,22 +1,169 @@
+  Button,
+  FormControl,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  TextareaAutosize,
+  Backdrop,
+  CircularProgress,
+  Grid
+} from "@mui/material";
+import { useState, useEffect } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import socket from "../utils/Socket.js"
 
 function MatchedRoom() {
+    const location = useLocation();
+    const matchEntry = location.state;
+    console.log("matchEntry: " + JSON.stringify(matchEntry))
+
+    const navigate = useNavigate();
+    const [roomInfo, setRoomInfo] = useState("")
+    const [isDialogOpen, setIsDialogOpen] = useState(false);
+    const [dialogTitle, setDialogTitle] = useState("");
+    const [dialogMsg, setDialogMsg] = useState("");
+    const [isConnected, setIsConnected] = useState(socket.connected);
+    const [isRoomCreated, setRoomCreated] = useState(false)
+    const [timeLeft, setTimeLeft] = useState(Number.MAX_VALUE)
+    const [isInterviewer, setIsInterviewer] = useState(true)
+
+    useEffect(() => {
+      // Requesting for collaboration service to create new room
+      if (isConnected) socket.emit('createRoom', matchEntry)
+    }, [isConnected, matchEntry])
+
+    useEffect(() => {
+      socket.on('connect', () => {
+        setIsConnected(true);
+      });
+
+      socket.on('disconnect', () => {
+        setIsConnected(false);
+      });
+
+      // Listening for roomCreated event emitted from collaboration service
+      socket.on('roomCreated', (roomInfo) => {
+        setRoomInfo(roomInfo);
+        setRoomCreated(true);
+        setTimeLeft(roomInfo.timeLeft)
+        setIsInterviewer(roomInfo.interviewer === sessionStorage.getItem("username"))
+      })
+
+      return () => {
+        socket.off('connect');
+        socket.off('disconnect');
+      };
+    }, []);
+
+    useEffect(() => {
+      // Reduce timeLeft by calling setTimeLeft function 1 every second (1000ms)
+      if (timeLeft === 0 && !roomInfo.hasSwitched) {
+        setDialogTitle("Times up!");
+        setDialogMsg("Would you like to switch roles and continue?")
+        setIsDialogOpen(true)
+      } else if (timeLeft === 0 && roomInfo.hasSwitched) {
+        setDialogTitle("Times up!");
+        setDialogMsg("Would you like to try another question with the same partner?")
+        setIsDialogOpen(true)
+      } else {
+        const timer =
+          timeLeft > 0 && setInterval(() => setTimeLeft(timeLeft - 1), 1000);
+        return () => clearInterval(timer);
+      }
+    }, [timeLeft]);
+
+    const exitClicked = () => {
+      setIsDialogOpen(true);
+      setDialogTitle("Are you sure you want to leave this room?");
+      setDialogMsg("All progress made will be lost.");
+    };
+
+    const handleExit = () => {
+      // TODO: Remove the pair's entry from collaboration service DB
+      // TODO: Remove the pair's entry from matching service DB
+      navigate('/home');
+    }
+
+    const handleSwitch = () => {
```

Comment:
Hey, everything looks good :) But was thinking if this should be handled by the session end page instead?

Reply:
Yup, my bad! I've moved it to a new page.